src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getUserById } from './services/userService';
+
+type Props = {
+  onSubmit: (todos: Todo) => void;
+};
+
+export const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC<Props> = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userValue, setUserValue] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const addTodo = (newTodo: Todo) => {
+    const todoToAdd: Todo = {
+      ...newTodo,
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+    };
+
+    setTodos(currentTodos => [...currentTodos, todoToAdd]);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserValue(+event.target.value);
+    setHasUserError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserValue(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setHasTitleError(!title.trim());
+    setHasUserError(!userValue);
+
+    if (!title.trim() || !userValue) {
+      return;
+    }
+
+    const newTodo = {
+      title,
+      userValue,
+      completed: false,
+      user: getUserById(userValue),
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      userId: userValue,
+    };
+
+    addTodo(newTodo);
+
+    reset();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form onSubmit={handleSubmit} action="/api/todos" method="POST">
```

Comment:
The `action` and `method` attributes are unnecessary in this form since the submission is handled by React. Consider removing them to avoid confusion.